sklearn/preprocessing/data.py

```diff
@@ -2244,18 +2245,20 @@ def _transform_col(self, X_col, quantiles, inverse):
 
         X_col[upper_bounds_idx] = upper_bound_y
         X_col[lower_bounds_idx] = lower_bound_y
-        # for forward transform, match the output PDF
+        # for forward transform, match the output distribution
         if not inverse:
             with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
-                X_col = output_distribution.ppf(X_col)
-            # find the value to clip the data to avoid mapping to
-            # infinity. Clip such that the inverse transform will be
-            # consistent
-            clip_min = output_distribution.ppf(BOUNDS_THRESHOLD -
-                                               np.spacing(1))
-            clip_max = output_distribution.ppf(1 - (BOUNDS_THRESHOLD -
-                                                    np.spacing(1)))
-            X_col = np.clip(X_col, clip_min, clip_max)
+                if output_distribution == 'normal':
+                    X_col = stats.norm.ppf(X_col)
+                    # find the value to clip the data to avoid mapping to
+                    # infinity. Clip such that the inverse transform will be
+                    # consistent
+                    clip_min = stats.norm.ppf(BOUNDS_THRESHOLD - np.spacing(1))
```

Comment:
Maybe we should clip the data when ``output_distribution == 'uniform'`` to avoid backward incompatibility and to keep ``inverse_transform`` consistent with ``transform``? though the difference seems small (1e-7)
(Actually I agree that we don't need to clip when ``output_distribution == 'uniform'``, but if we decide to do so, we'll need to update ``inverse_transform`` accordingly)

Reply:
gentle ping if you have time here :) @albertcthomas

Reply:
In fact I think that for the uniform distribution the clipping is done L2246
```python
X_col[upper_bounds_idx] = upper_bound_y
X_col[lower_bounds_idx] = lower_bound_y
```
which is done for `transform` and `inverse_transform`

Reply:
But we're talking about different things right? @albertcthomas 
```
import numpy as np
from sklearn.preprocessing import QuantileTransformer
rng = np.random.RandomState(0)
X = [[1], [2], [3], [4]]
qt = QuantileTransformer(n_quantiles=10, random_state=0)
qt.fit_transform(X)
```
Before your PR:
```
array([[9.99999998e-08],
       [3.33333333e-01],
       [6.66666667e-01],
       [9.99999900e-01]])
```
After your PR
```
array([[0.        ],
       [0.33333333],
       [0.66666667],
       [1.        ]])
```
Although the new version might be more reasonable.

Reply:
I'm even wondering whether we need ``clip_min`` and ``clip_max`` when output_distribution == 'normal'.

Reply:
the easiest solution IMO will be to avoid backward incompatibility (i.e., clip when `output_distribution == 'uniform'`) and open an issue to discuss whether we need `clip_min` and `clip_max`.

Reply:
Ah yes thanks for the clarification.

Reply:
for me it's a bug fix as there is no reason to clip for uniform distribution. Difference is also super tiny (order of float32 eps which is what is used for BOUND_THRESHOLDS). I am fine with the current PR as proposed.

Reply:
> for me it's a bug fix as there is no reason to clip for uniform distribution. Difference is also super tiny (order of float32 eps which is what is used for BOUND_THRESHOLDS). I am fine with the current PR as proposed.

Happy to regard it as a bug fix.
If so, we need to update inverse_transform accordingly. (around L2222)